MoreLinq/IgnoreErrors.cs

```diff
@@ -0,0 +1,244 @@
+        /// <param name="error1Predicate">
+        /// A function that receives the exceptions of type <typeparamref name="TException1"/>
+        /// that occurs during the iteration and indicates if it should be ignored or not.
+        /// </param>
+        /// <param name="error2Predicate">
+        /// A function that receives the exceptions of type <typeparamref name="TException2"/>
+        /// that occurs during the iteration and indicates if it should be ignored or not.
+        /// </param>
+        /// <param name="error3Predicate">
+        /// A function that receives the exceptions of type <typeparamref name="TException3"/>
+        /// that occurs during the iteration and indicates if it should be ignored or not.
+        /// </param>
+        /// <returns>
+        /// A sequence of elements <paramref name="source"/> that ignore the exceptions of type
+        /// <typeparamref name="TException1"/>, <typeparamref name="TException2"/> or <typeparamref name="TException3"/>
+        /// in accordance with predicates.
+        /// </returns>
+        /// <remarks>
+        /// This operator uses deferred execution and streams its results.
+        /// Exceptions are caught in order which predicates were passed, as well in a catch block.
+        /// </remarks>
+
+        public static IEnumerable<T> IgnoreErrors<T, TException1, TException2, TException3>(
+            this IEnumerable<T> source,
+            Func<TException1, bool> error1Predicate,
+            Func<TException2, bool> error2Predicate,
+            Func<TException3, bool> error3Predicate)
+            where TException1 : Exception
+            where TException2 : Exception
+            where TException3 : Exception
+        {
+            if (error1Predicate == null) throw new ArgumentNullException(nameof(error1Predicate));
+            if (error2Predicate == null) throw new ArgumentNullException(nameof(error2Predicate));
+            if (error3Predicate == null) throw new ArgumentNullException(nameof(error3Predicate));
+
+            return IgnoreErrorsImpl(source, error1Predicate, error2Predicate, error3Predicate);
+        }
+
+        static IEnumerable<T> IgnoreErrorsImpl<T, TException1, TException2, TException3>(
+            this IEnumerable<T> source,
+            Func<TException1, bool> error1Predicate,
+            Func<TException2, bool> error2Predicate,
+            Func<TException3, bool> error3Predicate)
+            where TException1 : Exception
+            where TException2 : Exception
+            where TException3 : Exception
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+
+            return _(); IEnumerable<T> _()
+            {
+                using (var e = source.GetEnumerator())
+                {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!e.MoveNext()) break;
```

Comment:
`break` on next line for consistency. It would have been okay on the same line as you have it now if you styled the `catch` blocks likewise, as in:

```c#
try { if (!e.MoveNext()) break; }
catch (TException1 ex) when (error1Predicate?.Invoke(ex) ?? true) { continue; }
catch (TException2 ex) when (error2Predicate?.Invoke(ex) ?? true) { continue; }
catch (TException3 ex) when (error3Predicate?.Invoke(ex) ?? true) { continue; }
```